### ISSUE TITLE
[no-empty-interface] Disable in declaration merging contexts

### ISSUE THREAD

**Issue by kripod** [typescript-eslint/typescript-eslint#1875]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-interface": "error"
  }
}
```

```TS
import { tokens } from './theme.treat';

declare module 'treat/theme' {
  type Tokens = typeof tokens;
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface Theme extends Tokens {}
}
```

The code above can be found [here](https://github.com/kripod/glaze/blob/08cc3fd809df7db2af68767f2c618a109b2a523c/packages/example-gatsby/src/gatsby-plugin-glaze/treat.d.ts), along with the referenced `theme.treat.ts` file in the same directory.

**Expected Result**

No errors should be thrown, as the `Theme` interface of `module 'treat/theme'` is [being augmented](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).

**Actual Result**

> An interface declaring no members is equivalent to its supertype.

(As thrown by `@typescript-eslint/no-empty-interface`.)

**Additional Info**

The code above gets auto-fixed to:

```ts
import { tokens } from './theme.treat';

declare module 'treat/theme' {
  type Tokens = typeof tokens;
  export type Theme = Tokens;
}
```

Which throws a TypeScript error `ts(2300)`:

> Duplicate identifier 'Theme'.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.12.0` |
| `yarn`                             | `1.22.4` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
This isn't really doable, because is that every single context is a declaration merging context.

I don't think disabling the rule entirely is the correct course of action, anyways - I'd rather have the rule knowingly and rarely false positive instead of false negative  
Narrowing the scope of this down - we could do something when we detect an interface is within an ambient declaration (i.e. is a child of `TSModuleDeclaration` with `declare: true`).

I think it would be a good idea to use a suggestion fixer instead of an auto fixer when we detect this case.

Happy to accept a PR to make this happen!

**Comment by vapurrmaid** [comment]
I took a crack at it from what I understand of the situation ^
